simulation/modules/plot.py

```diff
@@ -11,61 +13,85 @@
+        plot.macrophage_counts[step_num] = len(state.macrophage.cells.alive())
+        plot.neutrophil_counts[step_num] = len(state.neutrophil.cells.alive())
 
-        state.plot.step_num += 1
+        plot.step_num += 1
 
         return state
 
-    def finalize(self, state: State):
-        num_steps = state.plot.step_num
-        time_steps = state.plot.time_steps[:num_steps]
-        fungal_burdens = state.plot.fungal_burdens[:num_steps]
-        macrophage_counts = state.plot.macrophage_counts[:num_steps]
-        neutrophil_counts = state.plot.neutrophil_counts[:num_steps]
-        # define additional y_val here
+    def setup_plot(self, ylabel: str, title: str, filename: str):
+        plt.xlabel('Time Unit (Typically Hours)')
+        plt.ylabel(ylabel)
+        plt.title(title)
+        plt.grid(True)
+        plt.xlim(xmin=0)
+        plt.ylim(ymin=0)
+        plt.savefig(f'{filename}.png')
+        plt.clf()
 
+    def finalize(self, state: State):
+        plot_output_folder = self.config.get('plot_output_folder')
+        if not os.path.exists(plot_output_folder):
+            os.mkdir(plot_output_folder)
+        plot: PlotState = state.plot
+        num_steps = plot.step_num
+        time_steps = plot.time_steps[:num_steps]
         plt.figure()
-        plt.xlabel('Time')
-        plt.grid(True)
 
-        for y_val, y_label, title, filename in [
-            (fungal_burdens, 'Fungal Burden', 'Fungal Burden vs Time', 'fungal_burden_vs_time'),
-            (
-                macrophage_counts,
-                'Macrophage Count',
-                'Macrophage Count vs Time',
-                'macrophage_count_vs_time',
-            ),
-            (
-                neutrophil_counts,
-                'Neutrophil Count',
-                'Neutrophil Count vs Time',
-                'neutrophil_count_vs_time',
-            ),
-            # define additional plots here at 4-tuples
-        ]:
-            plt.clf()
-            plt.plot(time_steps, y_val)
-            plt.ylabel(y_label)
-            plt.title(title)
-            plt.savefig(f'{filename }.png')
+        fungal_burdens = plot.fungal_burdens[:num_steps]
+        plt.plot(time_steps, fungal_burdens, 'b.-')
+        self.setup_plot(
+            'Fungal Burden', 'Fungal Burden vs Time', plot_output_folder + 'fungal_burden_vs_time'
+        )
+
+        macrophage_counts = plot.macrophage_counts[:num_steps]
+        plt.plot(time_steps, macrophage_counts, 'b.-')
+        self.setup_plot(
+            'Macrophage Count',
+            'Macrophage Count vs Time',
+            plot_output_folder + 'macrophage_count_vs_time',
+        )
+
+        neutrophil_counts = plot.neutrophil_counts[:num_steps]
+        plt.plot(time_steps, neutrophil_counts, 'b.-')
+        self.setup_plot(
+            'Neutrophil Count',
+            'Neutrophil Count vs Time',
+            plot_output_folder + 'neutrophil_count_vs_time',
+        )
+
+        ax = plt.subplot(111)
```

Comment:
IMO, this triple-digit form of providing arguments is very unintuitive, and probably left over from making this a MATLAB-compatible API. I'd suggest using 3 separate arguments, as documented in [`*args` of `matplotlib.pyplot.subplot`](https://matplotlib.org/3.2.1/api/_as_gen/matplotlib.pyplot.subplot.html#matplotlib.pyplot.subplot).